File: src/com/prime/codility/lessons/cyclicrotation/cyclic_rotation.js
function solution (A, K){
    if ( A.length ===0) return A;

    var temp;
    for(let n=0; n<K; n++){
        temp = A[A.length - 1];
        for(let m=A.length - 1; m>0; m--){
            A[m] = A[m-1];
        }
        A[0] = temp;
    }
    return A;
}


//run
var A = [3, 8, 9, 7, 6]
var K = 3
var B = solution(A, K);
console.log(B);

File: src/com/prime/codility/lessons/cyclicrotation/CyclicRotation.java
// you can also use imports, for example:
// import java.util.*;

// you can write to stdout for debugging purposes, e.g.
// System.out.println("this is a debug message");

class Solution {
    public int[] solution(int[] A, int K) {
        if(K==A.length || A.length==0) return A;
        for(int k=0; k<K; k++) {
            int prev = A[A.length-1]; // hold last/previous value
            int temp; // hold current value for switching

            for(int i=0; i<A.length; i++) {
                temp = A[i];
                A[i] = prev;
                prev = temp;
            }
        }
        return A;
    }
}
